### ISSUE TITLE
Rapid clicking "Next" then "Previous" throws errors, breaks app

### ISSUE THREAD

**Issue by allthesignals** [NYCPlanning/labs-regional-viz#64]
Certain race conditions with mapbox gl's map instance being loaded and the framework attempting to update attributes lends itself to some unpredictable error states with the app...

In this case, if a user rapidly clicks "next" several times, then clicks "previous", there will be issues about missing sources.

**Comment by allthesignals** [comment]
The test passes, but you can see the errors in the console log:

<img width="1029" alt="screen shot 2018-03-01 at 11 24 55 am" src="https://user-images.githubusercontent.com/5004319/36856140-49092c3c-1d43-11e8-9713-20d16eba9145.png">

See branch for test: https://github.com/NYCPlanning/labs-regional-viz/tree/feature/failing-test-for-race-conditions

**Comment by andycochran** [comment]
I can't replicate this manually. Is the test clicking faster than humanly possible?

**Comment by allthesignals** [comment]
Yeah, you have to hold your mouse where the next button will be, then refresh. I know it's a reach but having race conditions in the app is bad because things may break unexpectedly and we won't know why!

**Comment by daragoldberg** [comment]
@allthesignals did we address this? can we close?
